### PR TITLE
Cypress Testing on CI

### PR DESCRIPTION
**Pull request by ksolo**
# Description

Adding Cypress tests with CI configurations to run tests on Travis.

Includes:
* ignoring artifacts from cypress run in gitignore (videos and screenshots)
* update travis yml to include the xvfb service and chrome addon
* update travis yml to call the cypress-run script
* cypress-run script handles additional configuration to setup ganache, db, static assets and also run the app and cypress tests
* update docker files to build chrome to run tests in a container
* add initial cypress tests

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -1 +1,5 @@
-{}
+{
+  "chromeWebSecurity": true,
+  "baseUrl": "http://localhost:8000",
+  "projectId": "e7rv3w"
```
Comment (chibie) [inline cypress.json:4]: is `projectId` a secret?

Comment (ksolo) [reply cypress.json:4]: No, it matches the test runs to the dashboard. There is not an ability to read data or access anything from this being exposed.

**Review by chibie — APPROVED** [review]

**Review by zacheryschiller — APPROVED** [review]

**Comment by thelostone-mc** [comment]
The commit history seems to have other commit not related this PR ! 
Any idea why that's there ? 

Would we need additional documentation to explain how to use/run this locally / is it not required?

**Comment by ksolo** [comment]
@thelostone-mc I can rebase this and remove the other commit. I'll add documentation for running tests locally as well.

**Pull request merged**